### PR TITLE
Add documentation on conditional retries

### PR DESCRIPTION
**Pull request by serinamarie**
Would love to get this into 1/4 release as the corresponding code is in.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] This pull request includes tests or only affects documentation.
- [x] This pull request includes a label categorizing the change e.g. `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed

### CODE REVIEW

File: docs/concepts/tasks.md [hunk 1]
```diff
@@ -268,6 +268,42 @@ def some_task_with_manual_backoff_retries():
    ...
 ```
 
+The `retry_condition_fn` option accepts a callable that returns a boolean. If the callable returns `True`, the task will be retried. If the callable returns `False`, the task will not be retried. The callable accepts three arguments &mdash; the task, the task run, and the state of the task run. The following task will retry on non-401 and non-404 HTTP status codes:
```
Comment (billpalombi) [inline docs/concepts/tasks.md:271]:
```suggestion
The `retry_condition_fn` option accepts a callable that returns a boolean. If the callable returns `True`, the task will be retried. If the callable returns `False`, the task will not be retried. The callable accepts three arguments &mdash; the task, the task run, and the state of the task run. The following task will retry on HTTP status codes other than 401 or 404:
```

**Review by billpalombi — APPROVED** [review]
I would rephrase the status code sentence, but otherwise lgtm

**Comment by N-Demir** [comment]
Hey! Maybe you've addressed this somewhere else already, and I'm not sure if you'll see this since the issue is closed, but this documentation doesn't work? Opened #11633
